### PR TITLE
Draft: Introduce github actions as pr check

### PR DESCRIPTION
**Pull request by odockal**
Signed-off-by: Ondrej Dockal <odockal@redhat.com>

# Pull Request Checklist
## General
* Is this a blocking issue or new feature? If yes, QE needs to +1 this PR

## Code
* Are method-/class-/variable-names meaningful?
* Are methods concise, not too long?
* Are catch blocks catching precise Exceptions only (no catch all)?

## Testing
* Are there unit-tests?
* Are there integration tests (or at least a jira to tackle these)?
* Is the non-happy path working, too?
* Are other parts that use the same component still working fine?

## Function
* Does it work?

### CODE REVIEW

File: .github/workflows/pr-check.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
+name: jbt-pr-check
+
+on:
+  push:
+    branches: [master]
```
Comment (jeffmaury) [inline .github/workflows/pr-check.yml:5]: To be removed

File: .github/workflows/pr-check.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+    # sonar flag are not included - requires adding secrets into repo security
+      run: mvn clean verify -U -fae -e -B -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true
+    - name: Archiving test artifacts
+      uses: actions/upload-artifact@v2
+      with: 
+        name: junit-logs
+        path: |
+          *tests/*/target/surefire-reports/*.xml
+          */*tests/*/target/surefire-reports/*.xml
+        retention-days: 5
```
Comment (jeffmaury) [inline .github/workflows/pr-check.yml:38]: Useless

File: .github/workflows/pr-check.yml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+    - name: Setup Java ${{ matrix.java }}
+      uses: actions/setup-java@v1
+      with:
+        java-version: ${{ matrix.java }}
+        java-package: jdk # (jre, jdk, or jdk+fx) - defaults to jdk
+        architecture: x64
+    - name: Build/Compile and run unit tests in repo using Maven step
+    # We do not have set settings.xml
+    # sonar flag are not included - requires adding secrets into repo security
+      run: mvn clean verify -U -fae -e -B -DskipITests=true -Dmaven.test.error.ignore=true -Dmaven.test.failure.ignore=true
```
Comment (jeffmaury) [inline .github/workflows/pr-check.yml:30]: Seems to me that we are missing settings.xml required by JBoss Tools

**Review by jeffmaury — CHANGES_REQUESTED** [review]

File: .github/workflows/pr-check.yml (hunk 3 above)
Comment (odockal) [reply .github/workflows/pr-check.yml:30]: @jeffmaury Yes we are, it is written in comment, I am curious if it works out of a box now. It is quite unfortunate that I need review approval for each change I plan to do without even testing it... This is suppose to be just test if we can live with PR check performed via Github Actions. I guess that I will find repo where I can experiment and return back here once it works flawlessly.

**Comment by odockal** [comment]
Closing in favor of https://github.com/jbosstools/jbosstools-openshift/pull/1989

**Pull request closed without merging**